### ISSUE TITLE
Trim whitespace on higher resolutions

### ISSUE THREAD

**Issue by SimenB** [facebook/docusaurus#4347]
## 💥 Proposal

When looking at documentation, there is a large swath of extraneous whitespace between the left hand ToC and the content which is not present on the right hand side.

![image](https://user-images.githubusercontent.com/1404810/109984295-e9740200-7d03-11eb-8dec-456ef1a6da15.png)

This is the `margin` added the the `container`.

![image](https://user-images.githubusercontent.com/1404810/109984356-fa247800-7d03-11eb-81a4-8c63a0cc28c4.png)

Notice how it almost hugs the ToC on the right hand side.

Reducing the size of the window makes the left hand side also hug, which IMO looks better

![image](https://user-images.githubusercontent.com/1404810/109984583-32c45180-7d04-11eb-954b-131307936367.png)

That also more closely resembles Docusarus v1, FWIW

![image](https://user-images.githubusercontent.com/1404810/109984638-42439a80-7d04-11eb-82f0-34a70f144246.png)

This looks extra weird when the entries in the ToC break due to not enough room when there's a bunch of (IMO) wasted space on the left

![image](https://user-images.githubusercontent.com/1404810/109984765-61dac300-7d04-11eb-946e-557933464584.png)

(example from https://jestjs.netlify.app/docs/api)

---

My suggestion is to cut most of the `margin` on the left side of the container. Short and sweet! 😀 

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yup!

**Comment by orta** [comment]
Yeah, I can see the tension here - that TOC throws off the centering for the main content. For large screens adding an offset for that extra whitespace would probably ease some of the 'this doesn't _feel_ centered'. 

Personally, I'm a fan of "if they want to make it obscenely wife, then let them" - which is why [the TypeScript website is remains aligned](https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html) and grows to whatever ridiculous widths you want to read it at. I get that there's generally [an optimal reading length](http://webtypography.net/2.1.2) but for long chunks of text I prefer to let folks make their own calls

**Comment by lex111** [comment]
Actually, we already increased the container size for large screen. However, we haven't applied this changes yet, but it should get better after that, see https://github.com/facebookincubator/infima/pull/62 for mode details.

| Before   | After    |
| -------- | -------- |
| ![image](https://user-images.githubusercontent.com/4408379/110002005-fefa2380-7d25-11eb-903c-45e013e530a0.png) | ![image](https://user-images.githubusercontent.com/4408379/110001936-eee24400-7d25-11eb-901d-a64b3460757a.png) |

**Comment by SimenB** [comment]
That's way better 👍 But there's still extra space on the left - any particular reason for that?

**Comment by lex111** [comment]
I can’t tell the exact reason because I didn't choose this layout, but I would assume that taking the container to full width will make the text harder to read (as mention above). A lot of docs sites limit the container width, so similar was done in Docusaurus as well.

**Comment by vamsi3** [comment]
Is there currently any way to customize the width of the main content by a user?

**Comment by SimenB** [comment]
>A lot of docs sites limit the container width, so similar was done in Docusaurus as well

Makes sense, but it looks sorta unbalanced when the left side ToC goes to the far left regardless of container width

**Comment by SimenB** [comment]
As an additional data point, this is not an issue on blog posts as the ToC hugs the content

![image](https://user-images.githubusercontent.com/1404810/110436280-54cb2480-80b4-11eb-9bd6-73efd34d1ec2.png)

**Comment by MateuszDabrowski** [comment]
Just bumped docusaurus version to 71 and saw the increased container size for a large screen and must say I really hope in the future version there will be an option to switch back to the previous spacing.

Sure, the new approach allows for more text to be displayed on the screen and less empty space, but at the same time, due to width, it makes reading much worse. 

There is a good reason why the press, Medium and other popular reading channels are using limited width column. It's huge for readability. Sometimes whitespace is good.

**Comment by slorber** [comment]
All this is quite opiniated, but remember that in any case you can add custom mediaqueries or customize the existing widths by overriding css variables.

Ideally I'd like to see if Facebook could provide us a design review or something because I'm not good enough in design to know what's best for Docusaurus 😅

**Comment by MateuszDabrowski** [comment]
I agree it is an opinionated topic, but I believe the consensus is that the recommended width for a block fo text is somewhere between 600 and 800px wide. I agree that it would be awesome to have a design review by someone that is an expert on the topic.

In the meantime, I did exactly what you suggested - overrode the CSS variables to limit the width and centre the content.

I also had to centre the doc pages without a navbar, as since alpha71 they were left-aligned (not sure whether this was done on purpose).

**Comment by HonkingGoose** [comment]
You could also take inspiration from TailwindCSS (and their Typography plugin) for how to format text nicely.

TailwindCSS v2 has a `max-w-prose` property that formats text nicely:

| Class | Properties |
| --- | --- |
| `max-w-prose` | `max-width`: `65ch` |

- https://tailwindcss.com/docs/max-width#prose
- https://github.com/tailwindlabs/tailwindcss-typography

**Comment by jetxr** [comment]
IMO

There is a _readability_ benefit in having the the content front and center with all sidebar away from the content at the edge of the screen.

However, for a docs site, there is a _usability_ benefit in having the sidebar next to the content area.

When a user is reading a docs site, reading multiple pages is a primary use-case for most. When having the sidebar next to the content, on a large screen, users won't have to move their eyes as much to discover topics, or move their cursor as much to use the navigation.

Visually having the content in prime focus, looks better. But when the actual use of a docs site is considered, it's beneficial to have navigation easily accessible.

For focused reading, sidebars can be collapsed, or a browser's reader view can be used.

![Screen Shot 2022-06-09 at 12 08 51 PM](https://user-images.githubusercontent.com/13171853/172781592-ed41a5cc-aeff-4326-9305-61fd83364e7e.png)

**Comment by slorber** [comment]
> However, for a docs site, there is a usability benefit in having the sidebar next to the content area.

It's possible to achieve that with swizzling and CSS, as demonstrated on the React-Native website:
- https://github.com/facebook/react-native-website
- https://reactnative.dev

<img width="1906" alt="CleanShot 2022-06-16 at 14 31 58@2x" src="https://user-images.githubusercontent.com/749374/174070316-16e9b005-3f06-4410-ba60-a78ab845d42d.png">

I do agree that we should do something on our site regarding this.

Particularly, for very large width gamer screens, which are more common these days, Docusaurus can look quite weird by default:

<img width="1871" alt="CleanShot 2022-06-16 at 14 32 45@2x" src="https://user-images.githubusercontent.com/749374/174070548-ae07a81d-c822-4af0-9c11-6ff35c6632b4.png">

**Comment by eric-horodyski** [comment]
Found this issue while looking for something similar. I'm looking to use Docusarus with CodeHike and build a similar layout to  [Stripe's Quickstart Guide](https://stripe.com/docs/checkout/quickstart) (which, I think is the ULTIMATE developer documentation layout) and am looking for ways to either remove whitespace on the right of the table of contents or remove them completely (conditionally, on certain pages). 

I totally agree that Docusaurus has the readability at heart with the current theme, but it would be nice to have additional information on how to adjust the theme with some common examples to help teach users how to Swizzle and adjust the layout.
